Automattic/gravatar
# Create package @gravatar/quick-editor

## Proposed Changes

This PR introduces a new package: `@gravatar/quick-editor`

Below is a usage example:

```HTML
<img
	class="avatar"
	src="https://1.gravatar.com/avatar/[AVATAR_HASH]?size=256"
/>
<button id="button">Edit Avatar</button>
```
```JavaScript
import GravatarQuickEditor, { GravatarQuickEditorCore } from '@gravatar/quick-editor';

document.addEventListener( 'DOMContentLoaded', () => {
	new GravatarQuickEditor( {
		email: 'email@provider.com',
		scope: [ 'avatars' ],
		editorTriggerSelector: '#button',
		avatarSelector: '.avatar',
	} );

	// QuickEditorCore usage
	const quickEditorCore = new GravatarQuickEditorCore( {
		email: 'email@provider.com',
		scope: [ 'avatars', 'about' ],
		local: 'es',
		onProfileUpdated: ( type: ProfileUpdatedType ) => {
			console.log( type );
		},
	} );

	document.querySelector( '#button' )?.addEventListener( 'click', () => {
		quickEditorCore.open();
	} );
} );
```

The constructor expects the following parameters:

* `triggerSelector`: A selector to the element that triggers the popup to be opened
* `email`: The user's email associated with the identity to be edited
* `options`: Optional parameter with the following options:
  * `scope`: allows defining the sections we want to show in the Quick Editor.
  * `avatarSelector`: A selector to match the avatar images on the page

## Testing Instructions

* Checkout this branch and run `npm run start`
* The playground page should open
* Change the files inside the folder `web/packages/quick-editor/playground/` with your Gravatar profile data
* Click the "Edit Avatar" the Quick Editor should open
* Change your Avatar image
* The images in the playground should change as well
